File: January/Java/Unique Number of Occurrences.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        HashMap<Integer, Integer> hashMap = new HashMap<>();
        for (int i = 0; i < arr.length; i++){
            if (hashMap.containsKey(arr[i])){
                hashMap.put(arr[i], hashMap.get(arr[i]) + 1);
            }
            else{
                hashMap.put(arr[i], 1);
            }
        }
        HashSet<Integer> hashSet = new HashSet<>();
        boolean flag = true;
        for (Map.Entry<Integer, Integer> mapElement : hashMap.entrySet()){
            if (!hashSet.add(mapElement.getValue())){
                flag = false;
            }
        }
        return flag;
    }
}

File: January/Python/Unique Number of Occurrences.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        nu_occurences = {}

        for interator in arr:
            nu_occurences[interator] = nu_occurences.get(interator, 0) + 1

        return len(set(nu_occurences.values())) == len(nu_occurences.keys())
        
